jaxley/io/swc.py

```diff
@@ -88,32 +96,40 @@ def _split_into_branches(
     current_branch = []
     all_types = []
 
-    # Loop over every line in the SWC file.
-    for c in content:
-        current_ind = c[0]  # First col is row_identifier
-        current_parent = c[-1]  # Last col is parent in SWC specification.
-        if current_parent == -1:
-            all_types.append(c[1])
-        else:
-            current_type = c[1]
+    # `previous_type` tracks the type (soma, axon...) of the previous SWC node.
+    previous_type = 0
+    for c in content:  # Loop over every line in the SWC file.
+        current_ind = int(c[0])  # First col is row_identifier
+        current_parent = int(c[-1])  # Last col is parent in SWC specification.
+        current_type = int(c[1])  # Second col is the type (soma, axon, basal,...).
```

Comment:
one could unpack in for-loop, but I guess less space for comments.

```
for idx, parent, type in content.tolist(): #tolist() casts to int
    ...
```

Reply:
I think it's fine to leave it as it is (nice for clarity)